### ISSUE TITLE
Extend inverse @ b as solve to batched dots

### ISSUE THREAD

**Issue by ricardoV94** [pymc-devs/pytensor#945]
This rewrite only triggers for non-blockwise dots, but there's no reason not to also cover Blockwise dots with a Blockise Solve (if solve is not blockwised by default already)

https://github.com/pymc-devs/pytensor/blob/ad27dc752798dfb75ed8d1c0a9b840859a8a898d/pytensor/tensor/rewriting/linalg.py#L90

**Comment by AdvH039** [comment]
@ricardoV94 I am  currently working on this issue and I have some doubts. As I see it the `solve` being used in the rewrite uses blockwise by default, so not much to do there. So all we have to do is trigger for the nodes that have 
`node.op = batched_variation_of_Dot | batched_variation of Dot 22` , use the current logic and write tests to see whether it is working or not, right?
And the batched variation of `Dot `we'll have to use from [here](https://github.com/pymc-devs/pytensor/blob/42e31c462cc49191f25226f0f3e22d945e223dc9/pytensor/tensor/math.py#L2745) right?
I could not find the corresponding one for `Dot22 `I guess we'll have to add it to `blas.py`.

**Comment by ricardoV94** [comment]
@AdvH039 you're right, the concern is that the rewrite currently doesn't apply with batch Dot. 

You don't have to worry about Batch Dot22 as that's a specialized op that shouldn't ever be batched (it is only introduced when there's no batching going on).

**Comment by AdvH039** [comment]
Hi @ricardoV94,
I have opened a recent topic in pymc discourse. I hope you can provide some inputs to it. I will try to complete the issue. Thanks!!

**Comment by ricardoV94** [comment]
> Hi [@ricardoV94](https://github.com/ricardoV94), I have opened a recent topic in pymc discourse. I hope you can provide some inputs to it. I will try to complete the issue. Thanks!!

I've missed it, it's better to keep the development discussion here on github.

**Comment by AdvH039** [comment]
@ricardoV94 Ok thanks. Please let me know if there are any provisions for 175hr and what ideas are you looking for as I want to start preparing the proposal as soon as possible.

**Comment by ricardoV94** [comment]
> [@ricardoV94](https://github.com/ricardoV94) I have started the discussion and introduced myself. Thanks!!

I meant the discussion for the issue should happen here, not for GSOC projects

**Comment by AdvH039** [comment]
Oh I'm sorry about that should I take it off? Thanks.(Also are there any developments regarding the gsoc project?)

**Comment by alessandrogentili001** [comment]
## Proposed approach for extending inv_as_solve to batched/Blockwise operations

Hi @ricardoV94 , I've been looking into this issue and I believe the fix is quite minimal. Here's my analysis and approach — I'd appreciate feedback before opening a PR.

### The Problem
The [inv_as_solve](https://github.com/pymc-devs/pytensor/blob/db7fa0792efa19f021296e166956acc4636e3c72/pytensor/tensor/rewriting/linalg.py#L136) rewriter currently only tracks Dot nodes. When a user writes pt.linalg.inv(A) @ b with batched (3D+) tensors, the @ operator produces a _matmul node (Blockwise(Dot)), not a plain Dot. Since _matmul isn't in the tracks list, the rewriter never fires for the batched case — meaning the numerically unstable explicit inverse is computed instead of using solve.

### The Fix
Add _matmul to the rewriter tracks: @node_rewriter([Dot, _matmul])
Remove the redundant isinstance(node.op, Dot) guard inside the function body, since it would reject _matmul nodes (which are 
Blockwise, not Dot). The tracks decorator already guarantees only matching nodes reach the function.
No changes to the replacement logic are needed — solve() already handles batched inputs natively.
